src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.Cache.cs

```diff
@@ -47,6 +49,8 @@ private static bool GetConfigurationCacheEnabled()
         public readonly List<SslApplicationProtocol> ApplicationProtocols;
         public readonly QUIC_ALLOWED_CIPHER_SUITE_FLAGS AllowedCipherSuites;
 
+        [UnscopedRef] private readonly ReadOnlySpan<byte> SettingsSpan => MemoryMarshal.AsBytes(new ReadOnlySpan<QUIC_SETTINGS>(in Settings));
```

Comment:
What is this saving us interms of actual perf improvement? How many peoply actually understand what `UnscopedRef` and this property code does and why is this important to have?

Reply:
It's about 1400x faster and allocates 2kB less per comparison:

<img width="532" alt="image" src="https://github.com/dotnet/runtime/assets/25316035/6668664f-a4e2-4e3f-b953-7540720122b9">


```csharp
using System.Runtime.InteropServices;
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;

BenchmarkRunner.Run<Benchmarks>();

[MemoryDiagnoser(false)]
public class Benchmarks
{
	private readonly QUIC_SETTINGS a = default;
	private readonly QUIC_SETTINGS b = default;

	[Benchmark]
	public bool Compare1()
	{
		return a.Equals(b);
	}

	[Benchmark]
	public bool Compare2()
	{
		return MemoryMarshal.AsBytes(new ReadOnlySpan<QUIC_SETTINGS>(in a)).SequenceEqual(MemoryMarshal.AsBytes(new ReadOnlySpan<QUIC_SETTINGS>(in b)));
	}
}
```

Reply:
The above is the worst case obviously, where the settings are completely equal.

Reply:
The original code (`Settings.Equals(other.Settings)`) used the default struct equality comparer which:
- boxes second argument
- makes a defensive copy of the first argument (since Equals is not a readonly method)
- potentially uses reflection-based comparison (probably not, since the type does not contain a reference, but I am not 100% sure).

This change eliminates the above and does not require us to keep track of new members.

As for `UnscopedRef`, the previous version had 

```cs
private readonly ReadOnlySpan<byte> SettingsSpan =>
    MemoryMarshal.AsBytes(new ReadOnlySpan<QUIC_SETTINGS>(ref Unsafe.AsRef(in Settings)));
```

`UnscopedRef` allows to get rid of Unsafe.AsRef and make it a bit shorter. The property is private to limit potential exposure to other code.

Reply:
The `UnscopedRef` is also ideal for correctness, otherwise you might accidentally leak the ref out of the struct and lifetime rules enforcement will fail you since you broke the rules.

Reply:
I think there might be some padding, e.g. between `MtuDiscoveryMissingProbeCount` and `DestCidUpdateIdleTimeoutMs` it seems like there'd be 3 bytes of padding - which would make the `SequenceEquals` approach incorrect.

If we check each field manually, we still get very similar performance numbers to the `SequenceEquals` approach:
<details>
<summary>
Code for Compare3
</summary>

```csharp
	[Benchmark]
	public bool Compare3()
	{
		return a.Anonymous1.IsSetFlags == b.Anonymous1.IsSetFlags
			&& a.MaxBytesPerKey == b.MaxBytesPerKey
			&& a.HandshakeIdleTimeoutMs == b.HandshakeIdleTimeoutMs
			&& a.IdleTimeoutMs == b.IdleTimeoutMs
			&& a.MtuDiscoverySearchCompleteTimeoutUs == b.MtuDiscoverySearchCompleteTimeoutUs
			&& a.TlsClientMaxSendBuffer == b.TlsClientMaxSendBuffer
			&& a.TlsServerMaxSendBuffer == b.TlsServerMaxSendBuffer
			&& a.StreamRecvWindowDefault == b.StreamRecvWindowDefault
			&& a.StreamRecvBufferDefault == b.StreamRecvBufferDefault
			&& a.ConnFlowControlWindow == b.ConnFlowControlWindow
			&& a.MaxWorkerQueueDelayUs == b.MaxWorkerQueueDelayUs
			&& a.MaxStatelessOperations == b.MaxStatelessOperations
			&& a.InitialWindowPackets == b.InitialWindowPackets
			&& a.SendIdleTimeoutMs == b.SendIdleTimeoutMs
			&& a.InitialRttMs == b.InitialRttMs
			&& a.MaxAckDelayMs == b.MaxAckDelayMs
			&& a.DisconnectTimeoutMs == b.DisconnectTimeoutMs
			&& a.KeepAliveIntervalMs == b.KeepAliveIntervalMs
			&& a.CongestionControlAlgorithm == b.CongestionControlAlgorithm
			&& a.PeerBidiStreamCount == b.PeerBidiStreamCount
			&& a.PeerUnidiStreamCount == b.PeerUnidiStreamCount
			&& a.MaxBindingStatelessOperations == b.MaxBindingStatelessOperations
			&& a.StatelessOperationExpirationMs == b.StatelessOperationExpirationMs
			&& a.MinimumMtu == b.MinimumMtu
			&& a.MaximumMtu == b.MaximumMtu
			&& a._bitfield == b._bitfield
			&& a.MaxOperationsPerDrain == b.MaxOperationsPerDrain
			&& a.MtuDiscoveryMissingProbeCount == b.MtuDiscoveryMissingProbeCount
			&& a.DestCidUpdateIdleTimeoutMs == b.DestCidUpdateIdleTimeoutMs
			&& a.Anonymous2.Flags == b.Anonymous2.Flags
			&& a.StreamRecvWindowBidiLocalDefault == b.StreamRecvWindowBidiLocalDefault
			&& a.StreamRecvWindowBidiRemoteDefault == b.StreamRecvWindowBidiRemoteDefault
			&& a.StreamRecvWindowUnidiDefault == b.StreamRecvWindowUnidiDefault;
	}
```

</details>

<img width="521" alt="image" src="https://github.com/dotnet/runtime/assets/25316035/2b49555a-a578-48e7-9704-f67ef789b42e">